### PR TITLE
Allow React 18 as peer dependency

### PR DESCRIPTION
**Pull request by jonasws**
## What kind of change does this PR introduce?

Updates `package.json` to allow react 18 as peer dependency, which is now stable.

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -25,8 +25,8 @@
     "build-storybook": "build-storybook"
   },
   "peerDependencies": {
-    "react": "^16.13.1 || ^17.0.1",
-    "react-dom": "^16.13.1 || ^17.0.1"
+    "react": "^16.13.1 || ^17.0.1 || 18",
+    "react-dom": "^16.13.1 || ^17.0.1 || 18"
```
Comment (MildTomato) [inline packages/react/package.json:29]:
```suggestion
"react": "^16.13.1 || ^17.0.1 || ^18.0.0",
"react-dom": "^16.13.1 || ^17.0.1 || ^18.0.0"
 ```

File: packages/react/package.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
     "build-storybook": "build-storybook"
   },
   "peerDependencies": {
-    "react": "^16.13.1 || ^17.0.1",
-    "react-dom": "^16.13.1 || ^17.0.1"
+    "react": "^16.13.1 || ^17.0.1 || 18",
```
Comment (MildTomato) [inline packages/react/package.json:28]:
```suggestion
"react": "^16.13.1 || ^17.0.1 || ^18.0.0",
 ```

**Review by MildTomato — CHANGES_REQUESTED** [review]
requires a changeset commit

**Comment by MildTomato** [comment]
instructions for setting changeset is here
https://github.com/supabase-community/auth-ui/blob/main/development.md#change-logs

**Review by MildTomato — APPROVED** [review]
added changeset

**Comment by jonasws** [comment]
Thanks for getting to this so quickly <3

**Pull request merged**